### DISCUSSION TITLE
Binding C-f, C-b to scroll through pages.

### DISCUSSION THREAD

**Discussion by tbdcit** [meow-edit/meow#611] [Q&A]
When I try to bind "C-f" , "C-b" to scroll forward and back through pages it seems to break "meow-left" and "meow-right" (which want to forward and back pages instead of characters).  What I tried was:

```lisp
(global-unset-key (kbd "C-f"))
(global-unset-key (kbd "C-b"))
(global-set-key (kbd "C-f") 'meow-page-down)
(global-set-key (kbd "C-b") 'meow-page-up)
```

I guess that binding "C-f" also reaps "forward-char"? What is the right way to do this?

**Comment by eshrh** [comment]
It's because meow-left and meow-right call "C-b" and "C-f". We assume it's pretty rare that the user is modifying default emacs commands like you are.

The only (?) way to do this is to define another key to map to forward, and then redefine `meow--kbd-forward-char`:
```
(global-set-key (kbd "H-f") 'forward-char)
(global-set-key (kbd "C-f") 'meow-page-down)
(setq meow--kbd-forward-char "H-f")
```
and the same for "C-b" too.

I wonder if we can modify all the kmacros to also accept commands, that way you don't have to define another key, you can just set the command it uses internally directly.

**Reply by tbdcit** [reply]
Thanks for the answer! 

> I wonder if we can modify all the kmacros to also accept commands, that way you don't have to define another key, you can just set the command it uses internally directly.

That sounds like a great idea. There are a few situations (like this one) where overriding the standard bindings can be very useful.
